src/xai_grok/grok.py

```diff
@@ -0,0 +1,223 @@
+import http
+import logging
+
+import pydantic
+import requests
+
+import xai_grok.errors as errors
+import xai_grok.providers as providers
+import xai_grok.schemas as schemas
+
+
+class Grok:
+
+    GROK_API_BASE_URL = "https://api.x.ai"
+
+    logger = logging.getLogger(__name__)
+
+    def __init__(
+        self,
+        api_key: str,
```

Comment:
Parameter name collision in `__init__` method where `requests` parameter shadows the imported `requests` module, potentially causing unexpected behavior.